tests/models_tests/test_weavenet.py

```diff
@@ -85,11 +86,53 @@ def test_backward_gpu(model, model_processed, data):
         [cuda.to_gpu(d) for d in data]
     model.to_gpu()
     model_processed.to_gpu()
-    gradient_check.check_backward(model, (atom_data, adj_data), y_grad,
-                                  atol=1e-1, rtol=1e-1)
-    gradient_check.check_backward(model_processed, (atom_data_processed,
-                                                    adj_data), y_grad,
-                                  atol=1e-1, rtol=1e-1)
+    gradient_check.check_backward(
+        model, (atom_data, adj_data), y_grad, atol=1e-1, rtol=1e-1)
+    gradient_check.check_backward(
+        model_processed, (atom_data_processed, adj_data), y_grad,
+        atol=1e-1, rtol=1e-1)
+
+
+def test_forward_cpu_graph_invariant(model, data):
+    atom_data, adj_data = data[1], data[2]
+    y_actual = cuda.to_cpu(model(atom_data, adj_data).data)
+
+    permutation_index = numpy.random.permutation(atom_size)
+    permute_atom_data = permute_node(atom_data, permutation_index)
+    permute_adj_data = adj_data.reshape(
+        batch_size, atom_size, atom_size, pair_feature_dim
+    ).astype(numpy.float32)
+    permute_adj_data = permute_adj(
+        permute_adj_data, permutation_index, axis=[1, 2])
+    permute_adj_data = permute_adj_data.reshape(
+        batch_size, atom_size * atom_size, pair_feature_dim
+    ).astype(numpy.float32)
+    permute_y_actual = cuda.to_cpu(model(
+        permute_atom_data, permute_adj_data).data)
+    assert numpy.allclose(y_actual, permute_y_actual, rtol=1.e-4, atol=1.e-6)
```

Comment:
It seems the only difference of `test_forward_cpu_graph_invariant` and `test_forward_cpu_processed_graph_invariant` are array os atoms to use (`data[0]` vs. `data[1]`) and `axis` option of `permute_node`. I would suggest to unify these two methods.